cmd/process-agent/main_common.go

```diff
@@ -109,18 +109,30 @@ func init() {
 func fixDeprecatedFlags() {
 	deprecatedFlags := []string{
 		// Global flags
-		"-config", "-ddconfig", "-sysprobe-config", "-pid", "-info", "-version", "-check",
+		"-config", "--config", "-sysprobe-config", "-pid", "-info", "-version", "-check",
 		// Windows flags
 		"-install-service", "-uninstall-service", "-start-service", "-stop-service", "-foreground",
 	}
 
+	replaceFlags := map[string]string{
+		"-config":  "--cfgpath",
+		"--config": "--cfgpath",
+	}
+
 	for i, arg := range os.Args {
 		for _, f := range deprecatedFlags {
 			if !strings.HasPrefix(arg, f) {
```

Comment:
nit: Do you mind adding a comment here of why you use `HasPrefix` vs a regular equality check. I saw you did this in your previous refactor as due to args having  additional parameters such as `--config=myconfig`

Maybe a test?

Reply:
Adding a test is a good idea, for the deprecated flags logic since it's more complex. And yes, the example you have is the reason for this being more complex than just `==`.